Automattic/newspack-plugin
# feat(memberships): overlay style for content gate

### All Submissions:






### Changes proposed in this Pull Request:

Implements overlay style options for the content gate page:

<img width="1237" alt="image" src="https://user-images.githubusercontent.com/820752/228948621-48744ec9-7974-4d8d-906c-c2a8fba8379f.png">

When selected, **the overlay appears once the beginning of the post content reaches half the viewport**. The settings for the number of visible paragraphs/more block do not change:

https://user-images.githubusercontent.com/820752/228651035-9f865350-5839-4b59-b681-99dbd79bbd65.mov

---

Overlay options are:

 - **Placement**: center or bottom
 - **Size (width)**: small (400px) , medium (600px), large (800px)

On mobile, the overlay will always be placed at the bottom and full-width. 

---

For later discussions: some ads are rendered behind this overlay and we should probably have a way around this. One possibility, for a separate PR, is to allow the publisher to tweak visible ad placements in gated content, just like it's done for regular pages/posts:

<img width="284" alt="image" src="https://user-images.githubusercontent.com/820752/228649882-c6bb807a-4d43-4038-8fcd-b4ea69d865f1.png">

### How to test the changes in this Pull Request:

1. Check out this branch and confirm you have the latest `newspack-blocks` (for [this](https://github.com/Automattic/newspack-blocks/pull/1411))
2. Edit the gate and change to "Overlay" and use the presented default settings
3. Place a donate block in the gate
4. Visit any restricted content, scroll down and confirm the gate appears as described above, in the center and 800px wide
5. Visit on mobile, scroll down, and confirm the gate is positioned at the bottom and full-width
6. Change the placement to "bottom", visit the page, and confirm the change on desktop
7. Tweak the size and confirm the editor changes as well,save, and confirm the change on the frontend
8. Click to donate and confirm the checkout takes over the overlay
9. Close the checkout and confirm the gate is back
10. Complete the membership flow and confirm the page is refreshed, the content is unlocked and the gate no longer appears

### Other information:
